server/src/main/java/org/elasticsearch/index/get/ShardGetService.java

```diff
@@ -201,9 +202,15 @@ private GetResult innerGetLoadFromStoredFields(String type, String id, String[]
 
             if (!fieldVisitor.fields().isEmpty()) {
                 fieldVisitor.postProcess(mapperService);
-                fields = new HashMap<>(fieldVisitor.fields().size());
+                nonMetaDataFields = new HashMap<>();
+                metaDataFields = new HashMap<>();
                 for (Map.Entry<String, List<Object>> entry : fieldVisitor.fields().entrySet()) {
-                    fields.put(entry.getKey(), new DocumentField(entry.getKey(), entry.getValue()));
+                    boolean isMetadataField = MapperService.isMetadataField(entry.getKey());
```

Comment:
looks like `isMetadataField` variable is never used

Reply:
done